PSReadLine/StringBuilderLinewiseExtensions.cs

```diff
@@ -72,6 +72,23 @@ internal static Range GetRange(this StringBuilder buffer, int lineIndex, int lin
                 endPosition - startPosition + 1
                 );
         }
+
+        /// <summary>
+        /// Returns true if the specified position
+        /// is on an empty logical line
+        /// </summary>
+        /// <param name="buffer"></param>
+        /// <param name="cursor"></param>
+        /// <returns></returns>
+        public static bool IsLogigalLineEmpty(this StringBuilder buffer, int cursor)
+        {
+            return 
+                buffer.Length == 0 ||
+                (cursor == buffer.Length && buffer[cursor - 1] == '\n') ||
+                (cursor > 0 && buffer[cursor] == '\n') ||
+                (cursor > 0 && buffer[cursor] == '\n' && cursor < buffer.Length - 1 && buffer[cursor - 1] == '\n')
```

Comment:
> `(cursor > 0 && buffer[cursor] == '\n' && cursor < buffer.Length - 1 && buffer[cursor - 1] == '\n')`

Is this line really needed? If `cursor > 0 && buffer[cursor] == '\n'` evaluates to `true`, this line will not be evaluated; if that expression evaluates to `false`, then this line will always return false.

Reply:
You are right, this line is redundant.